### PR TITLE
Use foreach in consoleAddUser.

### LINKED ISSUE #15286: Move GetNextUserGeneratePW inside loop
See  `GetNextUserGeneratePW` [here](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/scaffold-identity?view=aspnetcore-3.0&tabs=visual-studio#additional-resources)

```
public static async Task Initialize(IServiceProvider serviceProvider, string userList)
{
    using (var context = new ApplicationDbContext(
        serviceProvider.GetRequiredService<DbContextOptions<ApplicationDbContext>>()))
    {
        var userPW = GetNextUserGeneratePW(userList);   // MOVE INSIDE LOOP
        while (userPW.user != null)
        {
            var userID = await EnsureUser(serviceProvider, userPW.password, userPW.user);
            NotifyUser(userPW);
            userPW = GetNextUserGeneratePW(userList);   // FIX ME
        }
    }
}
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 804e24dd-7e30-8cf5-2c99-1ffacb4582f6
* Version Independent ID: 3e25547b-e2c9-6634-2c38-2a77207f17d4
* Content: [Scaffold Identity in ASP.NET Core projects](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/scaffold-identity?view=aspnetcore-3.0&tabs=visual-studio#additional-resources)
* Content Source: [aspnetcore/security/authentication/scaffold-identity.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/security/authentication/scaffold-identity.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by serpent5**
Fixes #15286.

cc @Rick-Anderson

I think this is the lowest impact change to make this work, but I do think we can do better. I know I've already said that... :smiley: 

[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/security/authentication/scaffold-identity?view=aspnetcore-3.0&branch=pr-en-us-15308&tabs=visual-studio#disable-register-page

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
How about something along the lines

```
bool userNotNull=true;

while(userNotNull){
...
 
userNotNull = userPW.user == null;

**Comment by Rick-Anderson** [comment]
Don't worry about low impact. I think you can come up with some elegant code.

**Comment by Rick-Anderson** [comment]
Actually, I think I like the original better. It's more robust.

**Comment by serpent5** [comment]
How about if I add a commit to this with my proposed solution? You can compare that with the other options and I can easily revert it. We can pick and choose parts you like, if you've got time for that (I know you have more work than you could ever need).

I don't mind spending the time on this and it being thrown away if that's the end result.

**Comment by Rick-Anderson** [comment]
The original being the one in master.
Yes, see if you can come up with something more elegant.

It would be something like

while( (userPW = GetNextUserGeneratePW(userList)  != null)

**Comment by serpent5** [comment]
Let me know if you'd like me to annotate the latest commit with what I was thinking.

File: aspnetcore/security/authentication/scaffold-identity/consoleAddUser/Data/SeedData.cs [hunk 1]
```diff
@@ -1,71 +1,54 @@
...
-using WebApplication1.Data;
 
 namespace ContactManager.Data
 {
     public static class SeedData
     {
         #region snippet
 
-        public static async Task Initialize(IServiceProvider serviceProvider, string userList)
+        public static async Task Initialize(IServiceProvider serviceProvider, List<string> userList)
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/scaffold-identity/consoleAddUser/Data/SeedData.cs]: Format this so we don't get a horizontal scroll bar

image

**Comment by Rick-Anderson** [comment]
Nice work!

**Pull request merged**